### PR TITLE
Changes to produce an unsigned Omni transaction.

### PR DESCRIPTION
**Pull request by bitsim-chris**

### CODE REVIEW

**Comment by msgilligan** [comment]
Thanks @bitsim-chris -- I'll look at this soon.

File: omnij-core/src/main/java/foundation/omni/tx/EncodeMultisig.java [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
     private static final int maxKeys = 3;  /* Redeemable key + 2 data keys */
     private static final int maxDataKeys = maxKeys - 1;
-    private static final int dustAmountInSatoshis = 100000;
+    private static final int dustAmountInSatoshis = 684;
```
Comment (msgilligan) [inline omnij-core/src/main/java/foundation/omni/tx/EncodeMultisig.java]: @dexX7 Could this be a problem for the integration tests?

File: omnij-core/src/main/java/foundation/omni/tx/OmniTxBuilder.java [hunk 2]
```diff
@@ -18,6 +19,7 @@
  */
 public class OmniTxBuilder {
     private final long stdTxFee = 10000;
+    private static final int dustAmountInSatoshis = 546;
```
Comment (msgilligan) [inline omnij-core/src/main/java/foundation/omni/tx/OmniTxBuilder.java]: @dexX7 Could this be a problem for the integration tests?

Comment (dexX7) [inline omnij-core/src/main/java/foundation/omni/tx/OmniTxBuilder.java]: With the default value for `-minrelayfee`, this number is fine.

File: omnij-core/src/main/java/foundation/omni/tx/EncodeMultisig.java (hunk 1 above)
Comment (dexX7) [inline omnij-core/src/main/java/foundation/omni/tx/EncodeMultisig.java]: If there are more than two public keys (i.e. three including the sender's key), or if the sender's public key is not compressed, then this value is too low, assuming default values for `-minrelayfee`. A list of default values can be found here: https://github.com/mastercoin-MSC/mastercore/commit/426592e287f79b37fbdddb5cad14ce9bf2ca1317

Bitcoin and Omni Core produce compressed public keys via `getnewaddress`, so the remaining question is: do the payloads that are embedded via `encode` fit into one public key, or do they require two?

Comment (msgilligan) [inline omnij-core/src/main/java/foundation/omni/tx/EncodeMultisig.java]: This commit c4d5071c05a451843a74034bb354996795bb4587 should be the correct solution.

**Comment by msgilligan** [comment]
Thanks @bitsim-chris !

The following 3 commits should together be the equivalent of this PR:

1. c4d5071c05a451843a74034bb354996795bb4587 - **Use getMinNonDustValue() to set Omni output amounts**
1. ab7886beb47561115fb02ae424c08d5632e7da57 - **bitcoinj MIN_NONDUST & MIN_TX_FEE in OmniTxBuilder**
1. 0931d2af00261d802fcec0e7a61244e4e3384b1e - **unsigned tx creation from PR**

I may make some further changes and possibly even change the API or maybe after a little more testing (and review by @dexX7 ) I'll just release it as **v0.3.1**.

If you could give it a try as of this commit and let me know if there are any issues, that would be very helpful.

**Comment by dexX7** [comment]
@msgilligan: the changes look good to me.

**Comment by msgilligan** [comment]
Closing -- **v0.3.1** is released. Thanks @bitsim-chris !

**Pull request closed without merging**